src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html

```diff
@@ -1,249 +1,190 @@
+        </ng-template>
+      </div>
 
+      <!-- Confirm password -->
+      <div class="form-item">
+        <cds-password-label for="confirmpassword"
+                            label="Confirm password..."
+                            [invalid]="!userForm.controls.confirmpassword.valid && userForm.controls.confirmpassword.dirty"
+                            [invalidText]="confirmpasswordError"
+                            i18n> Confirm password
+          <input cdsPassword
+                 type="password"
+                 placeholder="Confirm password..."
+                 id="confirmpassword"
+                 name="confirmpassword"
+                 formControlName="confirmpassword">
+        </cds-password-label>
+        <ng-template #confirmpasswordError>
+          <span class="invalid-feedback"
+                *ngIf="userForm.showError('confirmpassword', formDir, 'match')"
+                i18n>Password confirmation doesn't match the password.</span>
+          <span class="invalid-feedback"
+                *ngIf="userForm.showError('confirmpassword', formDir, 'required')"
+                i18n>This field is required.</span>
+        </ng-template>
+      </div>
+      <!-- Password expiration date -->
+      <div class="form-item"
+           *ngIf="!authStorageService.isSSO()">
+        <cd-date-time-picker [ngClass]="{'required': pwdExpirationSettings.pwdExpirationSpan > 0}"
+                             [control]="userForm.get('pwdExpirationDate')"
+                             name="Password expiration date"
+                             placeholder="Password expiration date"
+                             [hasTime]="false"
+                             [pwdExpirationDateValue]="pwdExpirationDateflag"
+                             i18n-name>
+        </cd-date-time-picker>
+        <span class="invalid-feedback"
+              *ngIf="userForm.showError('pwdExpirationDate', formDir, 'required')"
+              i18n>This field is required.
+        </span>
+      </div>
+      <!--Full Name-->
+      <div class="form-item">
+        <cds-text-label for="name"
+                        i18n> Full Name
+          <input cdsText
+                 type="text"
+                 placeholder="Full Name..."
+                 id="name"
+                 name="name"
+                 formControlName="name">
+        </cds-text-label>
+      </div>
+      <!-- Email -->
+      <div class="form-item">
+        <cds-text-label for="email"
+                        [invalid]="!userForm.controls.email.valid && userForm.controls.email.dirty"
+                        [invalidText]="emailError"
+                        i18n>
+          Email
+          <input cdsText
+                 type="text"
+                 placeholder="Email..."
+                 id="email"
+                 name="email"
+                 formControlName="email">
+        </cds-text-label>
+        <ng-template #emailError>
+          <span class="invalid-feedback"
```

Comment:
![image](https://github.com/user-attachments/assets/22c1d399-f5a9-4020-b0f5-7a358193ebeb)

Reply:
I don't think we should accept email without  (.com, .co.in etc)

Reply:
nice catch! I'd say you've discovered a bug in the angular source code itself I guess ;). While talking with @pujaoshahu I looked into the source code of angular email validator (because we are just using the validator from angular without any change) and [saw the regex they are using](https://github.com/angular/angular/blob/main/packages/forms/src/validators.ts#L140). I tested this one in a typescript playground and saw the same behavior you described.

```
const EMAIL_REGEXP =
  /^(?=.{1,254}$)(?=.{1,64}@)[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)*@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;

console.log(EMAIL_REGEXP.test('something@something'))

console.log >>> True
```
ideally this should return false if we want to fix it. And as a fellow open source contributor if either of you want you can raise it in the [angular community and raise a fix for it ](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#-found-a-bug)there and eventually we can benefit from it?

Reply:
Oh, interesting. 
This is something that has been raised before by other contributors https://github.com/angular/angular/issues/32788

It looks like angular uses the HTML5 validation so user@randomdomain is a valid address with no top level domain. Some more info on the address https://en.wikipedia.org/wiki/Email_address

Then is up to us to dice how to handle the email address and if we want to invalidate addresses without TLD.
Personally I would not change the current behaviour, mainly becuase I do not see any particular advantage to restrict the field as in the end we can not really verify if a given email is valid or invalid anyways, so in the end it will be up to the user.
And secondly, someone might be rightfully using myrandom@email and be an actual local address.

Reply:
As per [Wiki](https://en.wikipedia.org/wiki/Email_address#Examples) and [this](https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address) link which is being referred in the package form/validatos.ts file, this is a valid email!
So it comes down to, if we want to make it more robust we need to implement `pattern() `instead of just `Validators.email()`

Reply:
And just FYI, there have been some PR's already raised by fellow contributers and not sure if they did make it,
For example: [PR](https://github.com/angular/angular/issues/17296), and the PR description is crazy!

Seems like there was a voting process as well and it didn't make through.

Reply:
keeping as it is